compiler/rustc_passes/src/dead.rs

```diff
@@ -772,35 +744,24 @@ fn check_item<'tcx>(
                 .iter()
                 .filter_map(|def_id| def_id.as_local());
```

Comment:
Nits: this local should be named `associated_item_def_ids`, and if the impl is local, all its associated items must be too.